src/file/table/table-cell.ts

```diff
@@ -0,0 +1,194 @@
+import { XmlComponent, XmlAttributeComponent, IXmlableObject } from "file/xml-components";
+
+export enum BorderStyle {
+    SINGLE = "single",
+    DASH_DOT_STROKED = "dashDotStroked",
+    DASHED = "dashed",
+    DASH_SMALL_GAP = "dashSmallGap",
+    DOT_DASH = "dotDash",
+    DOT_DOT_DASH = "dotDotDash",
+    DOTTED = "dotted",
+    DOUBLE = "double",
+    DOUBLE_WAVE = "doubleWave",
+    INSET = "inset",
+    NIL = "nil",
+    NONE = "none",
+    OUTSET = "outset",
+    THICK = "thick",
+    THICK_THIN_LARGE_GAP = "thickThinLargeGap",
+    THICK_THIN_MEDIUM_GAP = "thickThinMediumGap",
+    THICK_THIN_SMALL_GAP = "thickThinSmallGap",
+    THIN_THICK_LARGE_GAP = "thinThickLargeGap",
+    THIN_THICK_MEDIUM_GAP = "thinThickMediumGap",
+    THIN_THICK_SMALL_GAP = "thinThickSmallGap",
+    THIN_THICK_THIN_LARGE_GAP = "thinThickThinLargeGap",
+    THIN_THICK_THIN_MEDIUM_GAP = "thinThickThinMediumGap",
+    THIN_THICK_THIN_SMALL_GAP = "thinThickThinSmallGap",
+    THREE_D_EMBOSS = "threeDEmboss",
+    THREE_D_ENGRAVE = "threeDEngrave",
+    TRIPLE = "triple",
+    WAVE = "wave",
+}
+
+interface ICellBorder {
+    style: BorderStyle;
+    size: number;
+    color: string;
+}
+
+class CellBorderAttributes extends XmlAttributeComponent<ICellBorder> {
+    protected xmlKeys = { style: "w:val", size: "w:sz", color: "w:color" };
+}
+
+class BaseTableCellBorder extends XmlComponent {
+    setProperties(style: BorderStyle, size: number, color: string) {
+        let attrs = new CellBorderAttributes({
+            style: style,
+            size: size,
+            color: color,
+        });
+        this.root.push(attrs);
+    }
+}
+
+export class TableCellBorders extends XmlComponent {
+    constructor() {
+        super("w:tcBorders");
+    }
+
+    public prepForXml(): IXmlableObject {
+        return this.root.length > 0 ? super.prepForXml(): '';
+    }
+
+    addTopBorder(style: BorderStyle, size: number, color: string) {
+        const top = new BaseTableCellBorder("w:top");
+        top.setProperties(style, size, color);
+        this.root.push(top);
+    }
+
+    addStartBorder(style: BorderStyle, size: number, color: string) {
+        const top = new BaseTableCellBorder("w:start");
+        top.setProperties(style, size, color);
+        this.root.push(top);
+    }
+
+    addBottomBorder(style: BorderStyle, size: number, color: string) {
+        const top = new BaseTableCellBorder("w:bottom");
```

Comment:
rename variable for better readability :)

Reply:
I used the names of the methods (start/end) not (left/right) because if we are in right-to-left tables then start would mean right :) Just to be consistent with how word and this library works.

Reply:
Perfect.